File: DTMS365-AppSource/DynamicsTMS365/Models/AMSAppointment.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Web;

namespace DynamicsTMS365.Models
{
	/// <summary>
	/// ASMApptOdrFlg
	/// </summary>
	public class ASMApptOdrFlg
	{
		/// <summary>
		/// Validation
		/// </summary>
		public AMSValidation Validation { get; set; }
		/// <summary>
		/// Appt
		/// </summary>
		public AMSAppointment Appt { get; set; }
		/// <summary>
		/// Ord
		/// </summary>
		public AMSOrder[] Ord { get; set; }
		/// <summary>
		/// Flag
		/// </summary>
		public Boolean Flag { get; set; }
	}
	/// <summary>
	/// Appointments class for SCH-Appointments
	/// </summary>
	public class AMSAppointment
	{
		/// <summary>
		/// Property AMSApptControl Control/ID as INT
		/// </summary>
		public int AMSApptControl { get; set; }
		/// <summary>
		/// Property AMSApptCompControl Control/ID as INT
		/// </summary>
		public int AMSApptCompControl { get; set; }
		/// <summary>
		/// Property AMSApptCarrierControl Control/ID as INT
		/// </summary>
		public int AMSApptCarrierControl { get; set; }
		/// <summary>
		/// AMSApptCarrierSCAC Name Property as STRING
		/// </summary>
		public string AMSApptCarrierSCAC { get; set; }
		/// <summary>
		/// AMSApptCarrierName Name Property as STRING
		/// </summary>
		public string AMSApptCarrierName { get; set; }
		/// <summary>
		/// AMSApptDescription Name Property as STRING
		/// </summary>
		public string AMSApptDescription { get; set; }
		/// <summary>
		/// AMSApptStartDate Property as DateTime
		/// </summary>
		public DateTime AMSApptStartDate { get; set; }
		/// <summary>
		/// AMSApptEndDate Property as DateTime
		/// </summary>
		public DateTime AMSApptEndDate { get; set; }
		/// <summary>
		/// AMSApptTimeZone Name Property as STRING
		/// </summary>
		public string AMSApptTimeZone { get; set; }
		/// <summary>
		/// Property AMSApptRecurrenceParentControl Control/ID as INT
		/// </summary>
		public int? AMSApptRecurrenceParentControl { get; set; }
		/// <summary>
		/// AMSApptRecurrence Name Property as STRING
		/// </summary>
		public string AMSApptRecurrence { get; set; }
		/// <summary>
		/// AMSApptActualDateTime Property as DateTime
		/// </summary>
		public DateTime? AMSApptActualDateTime { get; set; }
		/// <summary>
		/// AMSApptStartLoadingDateTime Property as DateTime
		/// </summary>
		public DateTime? AMSApptStartLoadingDateTime { get; set; }
		/// <summary>
		/// AMSApptFinishLoadingDateTime Property as DateTime
		/// </summary>
		public DateTime? AMSApptFinishLoadingDateTime { get; set; }
		/// <summary>
		/// AMSApptActLoadCompleteDateTime Property as DateTime
		/// </summary>
		public DateTime? AMSApptActLoadCompleteDateTime { get; set; }
		/// <summary>
		/// AMSApptModDate Property as DateTime
		/// </summary>
		public DateTime AMSApptModDate { get; set; }
		/// <summary>
		/// AMSApptModUser Name Property as STRING
		/// </summary>
		public string AMSApptModUser { get; set; }
		/// <summary>
		/// AMSApptNotes Name Property as STRING
		/// </summary>
		public string AMSApptNotes { get; set; }
		/// <summary>
		/// AMSApptDockdoorID Name Property as STRING
		/// </summary>
		public string AMSApptDockdoorID { get; set; }
		/// <summary>
		/// DockDoorName Name Property as STRING
		/// </summary>
		public string DockDoorName { get; set; }
		/// <summary>
		/// Property AMSApptStatusCode Control/ID as INT
		/// </summary>
		public int AMSApptStatusCode { get; set; }
		/// <summary>
		/// AMSApptLabel Name Property as STRING
		/// </summary>
		public string AMSApptLabel { get; set; }
		/// <summary>
		/// AMSApptHover Name Property as STRING
		/// </summary>
		public string AMSApptHover { get; set; }
		/// <summary>
		/// Property AMSApptOrderCount as INT
		/// </summary>
		public int AMSApptOrderCount { get; set; }
		/// <summary>
		/// CompAMSColorCodeSettingColorCode Name Property as STRING
		/// </summary>
		public string CompAMSColorCodeSettingColorCode { get; set; }
		



		private byte[] _AMSApptUpdated;
		/// <summary>
		/// AMSApptUpdate Property as STRING
		/// </summary>
		public string AMSApptUpdated
		{
			get
			{
				if (this._AMSApptUpdated != null)
				{

					return Convert.ToBase64String(this._AMSApptUpdated);

				}
				return string.Empty;
			}
			set
			{
				if (string.IsNullOrEmpty(value))
				{

					this._AMSApptUpdated = null;

				}

				else
				{

					this._AMSApptUpdated = Convert.FromBase64String(value);

				}

			}
		}
		/// <summary>
		/// setUpdated Method for setting Update RowVersion Number
		/// </summary>
		/// <param name="val"></param>
		public void setUpdated(byte[] val) { _AMSApptUpdated = val; }
		/// <summary>
		/// getUpdated Method for getting the Appointment Updated
		/// </summary>
		/// <returns></returns>
		public byte[] getUpdated() { return _AMSApptUpdated; }
	}
}

File: NGL.DataAssemblies/Ngl.FreightMaster.Data/DTO/AMS/AMSAppointment.vb
Imports Microsoft.VisualBasic
Imports System
Imports System.Collections.Generic
Imports System.Linq
Imports System.Text
Imports System.Runtime.Serialization
Imports Ngl.Core.ChangeTracker

Namespace DataTransferObjects
    <DataContract()> _
    Public Class AMSAppointment
        Inherits DTOBaseClass

#Region " Data Members"


        Private _AMSApptControl As Integer
        <DataMember()> _
        Public Property AMSApptControl() As Integer
            Get
                Return Me._AMSApptControl
            End Get
            Set(value As Integer)
                If ((Me._AMSApptControl = value) _
                   = False) Then
                    Me._AMSApptControl = value
                    Me.SendPropertyChanged("AMSApptControl")
                End If
            End Set
        End Property

        Private _AMSApptCompControl As Integer
        <DataMember()> _
        Public Property AMSApptCompControl As Integer
            Get
                Return _AMSApptCompControl
            End Get
            Set(value As Integer)
                If ((Me._AMSApptCompControl = value) _
                   = False) Then
                    Me._AMSApptCompControl = value
                    Me.SendPropertyChanged("AMSApptCompControl")
                End If
            End Set
        End Property

        Private _AMSApptCarrierControl As Integer
        <DataMember()> _
        Public Property AMSApptCarrierControl() As Integer
            Get
                Return Me._AMSApptCarrierControl
            End Get
            Set(value As Integer)
                If ((Me._AMSApptCarrierControl = value) _
                   = False) Then
                    Me._AMSApptCarrierControl = value
                    Me.SendPropertyChanged("AMSApptCarrierControl")
                End If
            End Set
        End Property

        Private _AMSApptCarrierSCAC As String
        <DataMember()> _
        Public Property AMSApptCarrierSCAC() As String
            Get
                Return Left(Me._AMSApptCarrierSCAC, 20)
            End Get
            Set(value As String)
                If (String.Equals(Me._AMSApptCarrierSCAC, value) = False) Then
                    Me._AMSApptCarrierSCAC = Left(value, 20)
                    Me.SendPropertyChanged("AMSApptCarrierSCAC")
                End If
            End Set
        End Property

        Private _AMSApptCarrierName As String
        <DataMember()> _
        Public Property AMSApptCarrierName() As String
            Get
                Return Left(Me._AMSApptCarrierName, 40)
            End Get
            Set(value As String)
                If (String.Equals(Me._AMSApptCarrierName, value) = False) Then
                    Me._AMSApptCarrierName = Left(value, 40)
                    Me.SendPropertyChanged("AMSApptCarrierName")
                End If
            End Set
        End Property

        Private _AMSApptDescription As String
        <DataMember()> _
        Public Property AMSApptDescription() As String
            Get
                Return Left(Me._AMSApptDescription, 20)
            End Get
            Set(value As String)
                If (String.Equals(Me._AMSApptDescription, value) = False) Then
                    Me._AMSApptDescription = Left(value, 20)
                    Me.SendPropertyChanged("AMSApptDescription")
                End If
            End Set
        End Property

        Private _AMSApptStartDate As System.Nullable(Of Date)
        <DataMember()> _
        Public Property AMSApptStartDate() As System.Nullable(Of Date)
            Get
                Return Me._AMSApptStartDate
            End Get
            Set(value As System.Nullable(Of Date))
                If (Me._AMSApptStartDate.Equals(value) = False) Then
                    Me._AMSApptStartDate = value
                    Me.SendPropertyChanged("AMSApptStartDate")
                End If
            End Set
        End Property

        Private _AMSApptEndDate As System.Nullable(Of Date)
        <DataMember()> _
        Public Property AMSApptEndDate() As System.Nullable(Of Date)
            Get
                Return Me._AMSApptEndDate
            End Get
            Set(value As System.Nullable(Of Date))
                If (Me._AMSApptEndDate.Equals(value) = False) Then
                    Me._AMSApptEndDate = value
                    Me.SendPropertyChanged("AMSApptEndDate")
                End If
            End Set
        End Property

        Private _AMSApptTimeZone As String
        <DataMember()> _
        Public Property AMSApptTimeZone() As String
            Get
                Return Left(Me._AMSApptTimeZone, 50)
            End Get
            Set(value As String)
                If (String.Equals(Me._AMSApptTimeZone, value) = False) Then
                    Me._AMSApptTimeZone = Left(value, 50)
                    Me.SendPropertyChanged("AMSApptTimeZone")
                End If
            End Set
        End Property

        Private _AMSApptRecurrenceParentControl As System.Nullable(Of Integer)
        <DataMember()> _
        Public Property AMSApptRecurrenceParentControl() As System.Nullable(Of Integer)
            Get
                Return Me._AMSApptRecurrenceParentControl
            End Get
            Set(value As System.Nullable(Of Integer))
                If (Me._AMSApptRecurrenceParentControl.Equals(value) = False) Then
                    Me._AMSApptRecurrenceParentControl = value
                    Me.SendPropertyChanged("AMSApptRecurrenceParentControl")
                End If
            End Set
        End Property

        Private _AMSApptRecurrence As String
        <DataMember()> _
        Public Property AMSApptRecurrence() As String
            Get
                Return Left(Me._AMSApptRecurrence, 1024)
            End Get
            Set(value As String)
                If (String.Equals(Me._AMSApptRecurrence, value) = False) Then
                    Me._AMSApptRecurrence = Left(value, 1024)
                    Me.SendPropertyChanged("AMSApptRecurrence")
                End If
            End Set
        End Property

        Private _AMSApptActualDateTime As System.Nullable(Of Date)
        <DataMember()> _
        Public Property AMSApptActualDateTime() As System.Nullable(Of Date)
            Get
                Return Me._AMSApptActualDateTime
            End Get
            Set(value As System.Nullable(Of Date))
                If (Me._AMSApptActualDateTime.Equals(value) = False) Then
                    Me._AMSApptActualDateTime = value
                    Me.SendPropertyChanged("AMSApptActualDateTime")
                End If
            End Set
        End Property

        Private _AMSApptStartLoadingDateTime As System.Nullable(Of Date)
        <DataMember()> _
        Public Property AMSApptStartLoadingDateTime() As System.Nullable(Of Date)
            Get
                Return Me._AMSApptStartLoadingDateTime
            End Get
            Set(value As System.Nullable(Of Date))
                If (Me._AMSApptStartLoadingDateTime.Equals(value) = False) Then
                    Me._AMSApptStartLoadingDateTime = value
                    Me.SendPropertyChanged("AMSApptStartLoadingDateTime")
                End If
            End Set
        End Property

        Private _AMSApptFinishLoadingDateTime As System.Nullable(Of Date)
        <DataMember()> _
        Public Property AMSApptFinishLoadingDateTime() As System.Nullable(Of Date)
            Get
                Return Me._AMSApptFinishLoadingDateTime
            End Get
            Set(value As System.Nullable(Of Date))
                If (Me._AMSApptFinishLoadingDateTime.Equals(value) = False) Then
                    Me._AMSApptFinishLoadingDateTime = value
                    Me.SendPropertyChanged("AMSApptFinishLoadingDateTime")
                End If
            End Set
        End Property

        Private _AMSApptActLoadCompleteDateTime As System.Nullable(Of Date)
        <DataMember()> _
        Public Property AMSApptActLoadCompleteDateTime() As System.Nullable(Of Date)
            Get
                Return Me._AMSApptActLoadCompleteDateTime
            End Get
            Set(value As System.Nullable(Of Date))
                If (Me._AMSApptActLoadCompleteDateTime.Equals(value) = False) Then
                    Me._AMSApptActLoadCompleteDateTime = value
                    Me.SendPropertyChanged("AMSApptActLoadCompleteDateTime")
                End If
            End Set
        End Property

        Private _AMSApptModDate As System.Nullable(Of Date)
        <DataMember()> _
        Public Property AMSApptModDate() As System.Nullable(Of Date)
            Get
                Return Me._AMSApptModDate
            End Get
            Set(value As System.Nullable(Of Date))
                If (Me._AMSApptModDate.Equals(value) = False) Then
                    Me._AMSApptModDate = value
                    Me.SendPropertyChanged("AMSApptModDate")
                End If
            End Set
        End Property

        Private _AMSApptModUser As String
        <DataMember()> _
        Public Property AMSApptModUser() As String
            Get
                Return Left(Me._AMSApptModUser, 100)
            End Get
            Set(value As String)
                If (String.Equals(Me._AMSApptModUser, value) = False) Then
                    Me._AMSApptModUser = Left(value, 100)
                    Me.SendPropertyChanged("AMSApptModUser")
                End If
            End Set
        End Property

        Private _AMSApptUpdated As Byte()
        <DataMember()> _
        Public Property AMSApptUpdated() As Byte()
            Get
                Return _AMSApptUpdated
            End Get
            Set(ByVal value As Byte())
                _AMSApptUpdated = value
            End Set
        End Property

        Private _AMSApptNotes As String
        <DataMember()> _
        Public Property AMSApptNotes() As String
            Get
                Return Left(Me._AMSApptNotes, 4000)
            End Get
            Set(value As String)
                If (String.Equals(Me._AMSApptNotes, value) = False) Then
                    Me._AMSApptNotes = Left(value, 4000)
                    Me.SendPropertyChanged("AMSApptNotes")
                End If
            End Set
        End Property
         
        Private _AMSApptDockdoorID As String
        <DataMember()> _
        Public Property AMSApptDockdoorID() As String
            Get
                Return Left(Me._AMSApptDockdoorID, 20)
            End Get
            Set(value As String)
                If (String.Equals(Me._AMSApptDockdoorID, value) = False) Then
                    Me._AMSApptDockdoorID = Left(value, 20)
                    Me.SendPropertyChanged("AMSApptDockdoorID")
                End If
            End Set
        End Property

        Private _AMSApptStatusCode As Integer
        <DataMember()> _
        Public Property AMSApptStatusCode As Integer
            Get
                Return _AMSApptStatusCode
            End Get
            Set(value As Integer)
                _AMSApptStatusCode = value
            End Set
        End Property

        Private _AMSApptLabel As String
        <DataMember()> _
        Public Property AMSApptLabel() As String
            Get
                Return Left(Me._AMSApptLabel, 500)
            End Get
            Set(value As String)
                If (String.Equals(Me._AMSApptLabel, value) = False) Then
                    Me._AMSApptLabel = Left(value, 500)
                    Me.SendPropertyChanged("AMSApptLabel")
                End If
            End Set
        End Property

        Private _AMSApptHover As String
        <DataMember()> _
        Public Property AMSApptHover() As String
            Get
                Return Left(Me._AMSApptHover, 4000)
            End Get
            Set(value As String)
                If (String.Equals(Me._AMSApptHover, value) = False) Then
                    Me._AMSApptHover = Left(value, 4000)
                    Me.SendPropertyChanged("AMSApptHover")
                End If
            End Set
        End Property


        Private _AMSApptOrderCount As Integer
        <DataMember()>
        Public Property AMSApptOrderCount As Integer
            Get
                Return _AMSApptOrderCount
            End Get
            Set(value As Integer)
                _AMSApptOrderCount = value
            End Set
        End Property

        'Added By LVV On 5/30/18 For v-8.3 TMS365 Scheduler
        Private _CompAMSColorCodeSettingColorCode As String
        <DataMember()>
        Public Property CompAMSColorCodeSettingColorCode() As String
            Get
                Return Left(Me._CompAMSColorCodeSettingColorCode, 10)
            End Get
            Set(value As String)
                If (String.Equals(Me._CompAMSColorCodeSettingColorCode, value) = False) Then
                    Me._CompAMSColorCodeSettingColorCode = Left(value, 10)
                    Me.SendPropertyChanged("CompAMSColorCodeSettingColorCode")
                End If
            End Set
        End Property

        Private _DockDoorName As String
        <DataMember()>
        Public Property DockDoorName() As String
            Get
                Return Left(Me._DockDoorName, 50)
            End Get
            Set(value As String)
                If (String.Equals(Me._DockDoorName, value) = False) Then
                    Me._DockDoorName = Left(value, 50)
                    Me.SendPropertyChanged("DockDoorName")
                End If
            End Set
        End Property

#End Region



#Region " Public Methods"
        Public Overrides Function Clone() As DTOBaseClass
            Dim instance As New AMSAppointment
            instance = DirectCast(MemberwiseClone(), AMSAppointment)
            Return instance
        End Function

#End Region
    End Class
End Namespace
